lib/treqs/checker.rb

```diff
@@ -20,6 +24,9 @@ def call(config_hash)
       .then { |keys_set| validate_required(keys_set) }
       .then { |keys_set| validate_allowed(keys_set) }
 
+    config_hash
+      .then { |request_hash| validate_http_methods_allowed(request_hash) }
+
     config_hash
```

Comment:
What if we change the _validate_ functions to receive a _Hash_ instead of a _Set_?
```ruby
config_hash
  .then { |hash| validate_(hash) }
  .then { |hash| validate_(hash) }
  .then { |hash| validate_(hash) }

Reply:
Internally, the _validate_ functions could handle the conversion from _Hash_ to _Set_ (since it's a more appropriate data structure for the diffing):
```ruby
def validate_(hash)
  keys_set = hash.keys.to_set
  ...

  hash
end
```